### PR TITLE
Percept/upgrade_types

### PR DESCRIPTION
**Pull request by MHMCD**
User story:
As a Virtual Human I want an upgrade_types percept
So that I can perceive which upgrades are available.
#20

The percept was rewritten to:
`upgrade_type/3 - upgrade_type(<typeID>, <sourceFunctionID>, <targetFunctionID>)`

Returns information about the type of upgrades that are available for buildings

`<typeID>` - ID of the upgrade type
`sourceFunctionID` - The function ID of the target
`targetFunctionID` - The function ID of the upgrade 

Since every typeID contains a maximum of 1 upgrade pair.

### CODE REVIEW

**Comment by tvleest** [comment]
Changes login.java?

**Comment by MHMCD** [comment]
the redundant changes have been removed

**Comment by tvleest** [comment]
approved. /RAMpestampers

File: environment/src/main/java/tygronenv/EntityEventHandler.java [hunk 1]
```diff
@@ -101,12 +102,15 @@ private void notifyListener1(Event event) throws EntityException {
...
 				break;
+			case UPGRADE_TYPES:
+				createPercepts(event.<ItemMap<UpgradeType>> getContent(MapLink.COMPLETE_COLLECTION), type);
 			case INDICATORS:
 				//Creates the indicator/3 percepts.
 				createPercepts(event.<ItemMap<Indicator>> getContent(MapLink.COMPLETE_COLLECTION), type);
 				break;
 			default:
 				System.out.println("WARNING. EntityEventHandler received unknown event:" + event);
+				return;
```
Comment (klikooo) [inline environment/src/main/java/tygronenv/EntityEventHandler.java:140]: Why did you add a `return`here?

**Comment by NickTUD** [comment]
Has conflicts as well, resolve them first

**Comment by NickTUD** [comment]
Still has conflicts

**Comment by JoshCode** [comment]
Conflicts were fixed. Since it's already past the 48 hour period, this will be merged this evening if there are no objections.

\- CodeFox [CF](http://github.com/CodeFoxNL)

**Comment by HarmGriffioen** [comment]
Agreed, merge if there are no objections

**Comment by MaxGroenenboom** [comment]
I noticed there are \* imports in the code? I think checkstyle doesn't accept those.

**Comment by JoshCode** [comment]
@MaxGroenenboom Checkstyle doesn't say anything about it, it succeeds.

https://travis-ci.org/CodeFoxNL/tygron/builds/133377348#L2875

But I do agree, \* imports are dirty code.

**Comment by JoshCode** [comment]
Will resolve #20

**Comment by MaxGroenenboom** [comment]
@JoshCode I know, that's what confuses me. I checked the checkstyle file and it does include a module for \* imports, and when I test a star import in my code checkstyle doesn't accept it ^^
So to summarize it, I think our checkstyle configuration doesn't accept it, and I don't understand why it does in your Travis build ^^
Maybe the checkstylediff script skips that file or line for some reason. Maybe because it was included in a conflict merge?

**Comment by levilime** [comment]
Hey there yeah the Checkstyle should absolutely not pass with the \* imports. There is an error, in build https://travis-ci.org/CodeFoxNL/tygron/builds/133377348#L2875 it is on line 2676, but it did not propagate as a build failure because it served as input for an other program. #54 will fix this behaviour.

Please merge #54 before this ticket, and checkstyle won't let your build pass until you adhere to our checkstyle rules.

File: environment/src/main/java/tygronenv/EisEnv.java [hunk 2]
```diff
@@ -1,185 +1,166 @@
...
 import eis.eis2java.exception.TranslationException;
 import eis.eis2java.translation.Java2Parameter;
 import eis.eis2java.translation.Parameter2Java;
 import eis.eis2java.translation.Translator;
-import eis.exceptions.ActException;
-import eis.exceptions.EntityException;
-import eis.exceptions.ManagementException;
-import eis.exceptions.NoEnvironmentException;
-import eis.exceptions.PerceiveException;
+import eis.exceptions.*;
```
Comment (klikooo) [inline environment/src/main/java/tygronenv/EisEnv.java]: There are still soms star imports, also on line 15

**Comment by JoshCode** [comment]
Checkstyle is passing now, though the star imports are still in.

https://travis-ci.org/CodeFoxNL/tygron/builds/133778279

**Comment by MaxGroenenboom** [comment]
@JoshCode But did you pull the new Checkstyle fix? #54

**Comment by JoshCode** [comment]
Yes

**Comment by MaxGroenenboom** [comment]
Looks like Travis indeed built with the new checkstyle fix, so I guess it is still broken? I see a.o. EisEnv.java:8 has been changed, but the checkstyle script only checks for `EisEnv.java: [30, 167, 168]`

**Comment by JoshCode** [comment]
In the wrong branch, I'm a jackass.

**Comment by MaxGroenenboom** [comment]
Lol the travis build was also from the wrong branch. I should've noticed that :P

**Comment by JoshCode** [comment]
All checkstyle for our code has been done. It still fails because it thinks we changed some code that was already present, which I'm not fixing.

```
[ERROR] src/main/java/tygronenv/EisEnv.java:[72,5] (javadoc) JavadocMethod: Missing a Javadoc comment.
[ERROR] src/main/java/tygronenv/EisEnv.java:[157,57] (javadoc) JavadocMethod: Expected @throws tag for 'EntityException'.
[ERROR] src/main/java/tygronenv/EisEnv.java:[165,25] (design) VisibilityModifier: Variable 'j2p' must be private and have accessor methods.
[ERROR] src/main/java/tygronenv/EisEnv.java:[170,25] (design) VisibilityModifier: Variable 'p2j' must be private and have accessor methods.
[ERROR] src/main/java/tygronenv/EntityEventHandler.java:[121] (sizes) LineLength: Line is longer than 120 characters (found 123).
[ERROR] src/main/java/tygronenv/EntityEventHandler.java:[121,76] (whitespace) GenericWhitespace: '>' is followed by whitespace.
```

**Comment by MaxGroenenboom** [comment]
I see they are caused by this merge: https://github.com/eishub/tygron/pull/41/commits/c77fe2abe82ba7fba41f7c3e099a4013211c8f4b
I don't know what happened there, but I do know we can't merge a PR that would break the Travis build...

**Comment by MHMCD** [comment]
All checkstyle errors have been fixed and the Travis build is now passing.

**Comment by JoshCode** [comment]
Proof of passing: https://travis-ci.org/CodeFoxNL/tygron/builds/133808952

Checkstyle errors were the only complaint, which has been resolved, and the 48 hour comment period has passed. Merging.

\- CodeFox [CF](http://github.com/CodeFoxNL)
